probe-rs-target/src/chip.rs

```diff
@@ -58,6 +58,8 @@ pub struct Chip {
     /// ref: `<https://open-cmsis-pack.github.io/Open-CMSIS-Pack-Spec/main/html/sdf_pg.html#sdf_element_scanchain>`
     #[serde(default)]
     pub scan_chain: Option<Vec<ScanChainElement>>,
+    /// The default format for this chip
+    pub default_format: Option<String>,
```

Comment:
This shouldn't be a String, but a concrete type.

Reply:
And `default_format` feels a bit ambiguous, I think something like `default_binary_format` makes it a bit clearer what is meant here.

Reply:
I think this should not be an option and have a default in case its not set. Presumably ELF.

Reply:
I did have this initially, however, because probe-rs depends on probe-rs-target, probe-rs-target can't depend on probe-rs meaning I had to duplicate the `Format` enum in two places, and then add conversions for both. I'll add back the enum.

Reply:
Actually, the enum makes way more sense, and it is distinctly different to the `Format` enum in probe-rs. `Format` describes the input formats probe-rs understands, `BinaryFormat` describes the binary layout of a program on a given target.